frontend/src/bundles/common/hooks/use-app-theme/use-app-theme.hook.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from '~/bundles/common/hooks/hooks.js';
+import { storage } from '~/framework/storage/storage.js';
+
+type Theme = 'dark' | 'light';
+
+function useDarkTheme(): [
+    Theme | null,
+    React.Dispatch<React.SetStateAction<Theme | null>>,
+    boolean,
+] {
+    const [theme, setTheme] = useState<Theme | null>(null);
+    const [loading, setLoading] = useState(true);
+    useEffect(() => {
+        const fetchTheme = async (): Promise<void> => {
+            try {
+                const currentTheme = await storage.get('theme');
+                const colorTheme =
+                    currentTheme === null ? 'dark' : (currentTheme as Theme);
+                setTheme(colorTheme);
+            } catch (error) {
+                throw new Error(String(error));
+            } finally {
+                setLoading(false);
+            }
+        };
```

Comment:
let's add actions and store for this, theme will be retrieved from local storage on init, but then source of truth will be redux store.